app/components/cms_rich_text_block_component.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+class CmsRichTextBlockComponent < ViewComponent::Base
+  def build(obj)
+    klass =
+      case obj
+      in { type: "paragraph" } then Paragraph
+      in { type: "heading" } then Heading
+      in { type: "text" } then Text
+      in { type: "link" } then Link
+      in { type: "list" } then List
+      in { type: "list-item" } then ListItem
+      in { type: "image" } then Image
+      in { type: "quote"} then Quote
+      end
+
+    klass.new(blocks: obj)
+  end
+
+  erb_template <<~ERB
+    <div class="govuk-width-container cms-rich-text-block-component">
+      <div class="govuk-main-wrapper">
+        <div class="govuk-grid-row">
+          <div class="govuk-grid-column-two-thirds">
+            <% @obj.each do |child| %>
+              <%= render build(child) -%>
+            <% end %>
+          </div>
+        </div>
+      </div>
+    </div>
+  ERB
+
+  def initialize(blocks:)
+    @obj = blocks
+  end
+
+  class Paragraph < CmsRichTextBlockComponent
+    erb_template <<~ERB
+      <p class="govuk-body-m"><%- @obj[:children].each do |child| -%><%= render build(child) -%><%- end -%></p>
```

Comment:
Could this be put on multiple lines please.

Is there also a reason why the end tag is `<%- end -%>` rather than `<% end %>`?